spec/forms/applicants/basic_details_form_spec.rb

```diff
@@ -65,6 +65,18 @@
       end
     end
 
+    context 'with an invalid email' do
+      before { attributes[:email] = 'not-a-valid-email' }
```

Comment:
**[not a blocker/nit pick]** Consider re-defining the `attributes` variable instead of using before blocks, e.g.:

```ruby
let(:attributes) { attributes_for(:applicant).merge(email: 'not-a-valid-email') }
```

In my experience, I find it nicer to read and more manageable.